src/mrdsrg-spin-integrated/dsrg_mrpt2.cc

```diff
@@ -1735,23 +1730,25 @@ double DSRG_MRPT2::compute_energy_relaxed() {
             }
 
             // compute permanent dipoles
-            if (options_.get_bool("FORM_MBAR3")) {
+            if (foptions_->get_bool("FORM_MBAR3")) {
                 dm_relax = fci_mo.compute_ref_relaxed_dm(Mbar0_, Mbar1_, Mbar2_, Mbar3_);
             } else {
                 dm_relax = fci_mo.compute_ref_relaxed_dm(Mbar0_, Mbar1_, Mbar2_);
             }
         }
-    } else if (options_.get_str("CAS_TYPE") == "ACI") {
+    } else if (foptions_->get_str("CAS_TYPE") == "ACI") {
 
-        AdaptiveCI aci(reference_wavefunction_, options_, ints_, mo_space_info_);
-        aci.set_fci_ints(fci_ints);
+        auto state = make_state_info_from_psi_wfn(ints_->wfn());
+        AdaptiveCI aci(state, scf_info_, foptions_, mo_space_info_, fci_ints);
 
         Erelax = aci.compute_energy();
     } else {
-        // it is simpler here to call FCI instead of FCISolver
-        FCI fci(reference_wavefunction_, options_, ints_, mo_space_info_, fci_ints);
-        fci.set_max_rdm_level(1);
-        Erelax = fci.compute_energy();
+        auto state = make_state_info_from_psi_wfn(ints_->wfn());
+        auto fci =
+            make_active_space_solver("FCI", state, scf_info_, mo_space_info_, ints_, foptions_);
+        fci->set_max_rdm_level(1);
+        fci->set_active_space_integrals(fci_ints);
+        Erelax = fci->compute_energy();
     }
```

Comment:
Not needed immediately, but we could avoid all of this logic by calling the base class

Reply:
Yes, this can wait. By the way, ActiveSpaceSolver does not build the integral object anymore. I decided to decouple the two to make things more transparent and customizable. All the logic for making ActiveSpaceIntergral is handled in `make_active_space_solver` so user interactions with these classes is still simple.